src/mlpack/methods/ann/gan/gan_impl.hpp

```diff
@@ -325,14 +364,8 @@ EvaluateWithGradient(const arma::mat& /* parameters */,
     gradientGenerator *= multiplier;
   }
 
-  counter++;
   currentBatch++;
```

Comment:
So, I am thinking that this comment is also a bit out of scope here, but my intuition here is that we are holding `currentBatch` specifically so that we can track when `currentBatch % generatorUpdateStep == 0` so that we can pass the error from the discriminator network to the generator network.  If that's right, I think that the code could be simplified with ensmallen callbacks (so that we can just receive a callback when the number of steps is right, or something like that).  Like I said, probably we don't need to do that in this PR, but maybe it's a good idea to open an issue for it.  Let me know what you think (and sorry for the out of scope comment).

Reply:
I am not sure about `call backs`. But yes I will look into it and I will open an issue if it is achievable through `ensmallen callbacks`. 
Basically you are right about it. Most of the times users choose to train `the discriminator` twice before training  `the generator`. So `currentBatch` and `generatorUpdateStep` as you said are used for it.